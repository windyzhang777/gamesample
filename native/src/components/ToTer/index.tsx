// import {useDispatch, useSelector} from 'react-redux';
import React from 'react';
// import {chooseTreat, chooseTrick} from 'shared/store/trick-or-treat/actions';
import {View, Button, Text, StyleSheet} from 'react-native';
import ToterShared, {ToterProps} from 'shared/containers/Toter';
import {useDispatch} from 'react-redux';

const renderCandy = (cachedImages, candy) => {
  const items = [];

  // for (let step = 0; step < candy.count; step++) {
  //   items.push(
  //     <Image
  //       source={{uri: cachedImages[`${candy.id}.jpg`]}}
  //       alt={candy.id}
  //       key={`${candy.id}-${step}`}
  //     />,
  //   );
  // }

  items.push(
    <Text>
      {candy.id}: {candy.count}
    </Text>,
  );

  return items;
};

const ToTer = (props: ToterProps) => {
  // const dispatch = useDispatch();
  // const cachedImages = useSelector((state) => state.content);

  return (
    <ToterShared
      {...props}
      useDispatch={useDispatch}
      renderComponent={({
        chooseTrick,
        chooseTreat,
        id,
        cachedImages,
        candies,
        name,
      }: ToterProps) => {
        return (
          <View>
            <Text>Candy In {name}'s Pillowcase</Text>
            {candies &&
              candies.map((candy) => (
                <View key={candy.id}>{renderCandy(cachedImages, candy)}</View>
              ))}
            <View style={styles.buttonRow}>
              <Button onPress={() => chooseTrick()} title="Trick" />
              <Text>or</Text>
              <Button onPress={() => chooseTreat(id)} title="Treat" />
            </View>
          </View>
        );
      }}
    />
  );
};

const styles = StyleSheet.create({
  buttonRow: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
  },
  candyImage: {
    height: 20,
    width: 20,
  },
});

export default ToTer;
