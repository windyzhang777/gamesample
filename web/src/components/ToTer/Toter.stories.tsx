import React from 'react';
import {storiesOf} from '@storybook/react';
import Toter from '.';
import Provider from '../../utils/Provider';

storiesOf('Toter', module)
  .addDecorator((story) => <Provider story={story()} />)
  .add('With name and candy', () => {
    return (
      <Toter
        cachedImages={[]}
        candies={[
          {id: 1, count: 10},
          {id: 2, count: 100},
          {id: 3, count: 1337},
          {id: 4, count: 42},
        ]}
        name={'Kiana'}
        id={'10'}
        chooseTreat={() => {
          console.log('chooseTreat Pressed');
        }}
        chooseTrick={() => {
          console.log('chooseTrick Pressed');
        }}
      />
    );
  })
  .add('With long name no candy', () => {
    return (
      <Toter
        cachedImages={[]}
        candies={[]}
        name={'Kiana daughter of skyler'}
        id={'10'}
        chooseTreat={() => {
          console.log('chooseTreat Pressed');
        }}
        chooseTrick={() => {
          console.log('chooseTrick Pressed');
        }}
      />
    );
  })
  .add('With long name and lots of candy', () => {
    return (
      <Toter
        cachedImages={[]}
        candies={[
          {id: 1, count: 10},
          {id: 2, count: 100},
          {id: 3, count: 1337},
          {id: 4, count: 42},
        ]}
        name={'Kiana daughter of skyler'}
        id={'10'}
        chooseTreat={() => {
          console.log('chooseTreat Pressed');
        }}
        chooseTrick={() => {
          console.log('chooseTrick Pressed');
        }}
      />
    );
  });
