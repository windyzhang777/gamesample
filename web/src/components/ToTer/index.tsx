import {useDispatch} from 'react-redux';
import React from 'react';
import ToterShared, {ToterProps} from 'shared/containers/Toter';

const renderCandy = (cachedImages: any, candy: any) => {
  const items = [];

  // for (let step = 0; step < candy.count; step++) {
  //   items.push(
  //     <img src={cachedImages[`${candy.id}.jpg`]} alt={candy.id} key={`${candy.id}-${step}`} />,
  //   );
  // }

  items.push(
    <span>
      {candy.id}: {candy.count}
    </span>,
  );

  return items;
};

const ToTer = (props: ToterProps) => {
  return (
    <ToterShared
      {...props}
      useDispatch={useDispatch}
      renderComponent={({
        chooseTrick,
        chooseTreat,
        cachedImages,
        candies,
        name,
        id,
      }: ToterProps) => {
        return (
          <div>
            Candy In {name}'s Pillowcase
            {candies &&
              candies.map((candy: any) => (
                <div key={candy.id}>{renderCandy(cachedImages, candy)}</div>
              ))}
            <button onClick={() => chooseTrick()}>Trick</button>
            or
            <button onClick={() => chooseTreat(id)}>Treat</button>
          </div>
        );
      }}
    />
  );
};

export default ToTer;
