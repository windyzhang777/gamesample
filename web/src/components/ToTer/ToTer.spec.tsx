import React from 'react';
import {shallow} from 'enzyme';
import ToTer from './index';

const render = (props: any) => shallow(<ToTer {...props} />);

describe('ToTer Component', () => {
  it('renders with default props', () => {
    expect(render({})).toMatchSnapshot();
  });
});
