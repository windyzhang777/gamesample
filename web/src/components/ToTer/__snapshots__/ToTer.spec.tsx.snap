// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToTer Component renders with default props 1`] = `
<Toter
  renderComponent={[Function]}
  useDispatch={[Function]}
/>
`;
